applications/IgaApplication/custom_utilities/node_curve_geometry_3d.h

```diff
@@ -0,0 +1,206 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Thomas Oberbichler
+//
+
+#if !defined(KRATOS_NODE_CURVE_GEOMETRY_3D_H_INCLUDED)
+#define KRATOS_NODE_CURVE_GEOMETRY_3D_H_INCLUDED
+
+// System includes
+
+// External includes
+#include <ANurbs/Core>
+
+// Project includes
+#include "includes/define.h"
+#include "includes/Node.h"
+#include "includes/variables.h"
+
+namespace Kratos {
+
+/// Spatial NURBS curve geometry with Kratos Nodes as control points.
+/** Unlike the ANurbs::CurveGeometry, this class does not use static Points as
+ *  control points but Finite Element Nodes. That means that the Geometry
+ *  changes whenever the Nodes are moving.
+ */
+class NodeCurveGeometry3D
+    : public ANurbs::CurveGeometryBase<double, Kratos::array_1d<double, 3>>
+{
+protected:
+    using NodePointer = typename Node<3>::Pointer;
+
+public:
+    using CurveGeometryBaseType = ANurbs::CurveGeometryBase<double,
+        Kratos::array_1d<double, 3>>;
+    using typename CurveGeometryBaseType::KnotsType;
+    using typename CurveGeometryBaseType::ScalarType;
+    using typename CurveGeometryBaseType::VectorType;
+ 
+protected:
+    std::vector<NodePointer> mNodes;
+
+public:
+    /** Creates a new NodeCurveGeometry3D.
+     *
+     *  \param Degree Degree of the curve
+     *  \param NumberOfNodes Number of nodes
+     */
+    NodeCurveGeometry3D(
+        const int Degree,
+        const int NumberOfNodes)
```

Comment:
is this an integer or an unsigned integer? (std::size_t)

Reply:
It's an integer because `(evilness_of(unsigned) > evilness_of(using)) == true`

Reply:
I take your argument, but errors can happen copying from one to the other

Reply:
At least for absurdly huge integers

Reply:
Thats right! I also think that the range of the data type should match. Maybe `long`? Anyway: I don't think that someone will create a curve with more than 2.147.483.647 control points so soon. Meanwhile I would think a little bit about the right type.